components/support/error/macros/src/lib.rs

```diff
@@ -60,30 +60,30 @@ mod argument;
 #[proc_macro_attribute]
 pub fn handle_error(args: TokenStream, input: TokenStream) -> TokenStream {
     let args = syn::parse_macro_input!(args as syn::AttributeArgs);
-    let parsed = syn::parse_macro_input!(input as syn::Item);
-    TokenStream::from(match impl_handle_error(&parsed, &args) {
-        Ok(res) => res,
-        Err(e) => e.to_compile_error(),
-    })
+    match argument::validate(&args) {
```

Comment:
if we keep the `validate` in the `impl_handle_error` function, then we can get rid of this match block!

any error would be caught by the inner `match` and converted to a compile error

Reply:
I got a bit lazy there - I moved that around in a failed attempt to reuse this in another macro, but didn't change it back.